### ISSUE TITLE
Question: OC CLI: new-app - possible to pass environment values to build in S2I?

### ISSUE THREAD

**Issue by gt-tech** [openshift/origin#8795]
This is more of a question, I create an application using "oc new-app git://........." command and in this command i pass several environment variables using -e flag and all of those environment variables are made available to runtime container but not to builder.

For now I have defined the build time environments in .s2i/environment at root of repository but in our case this won't work as depending on which particular module is being build (we use MAVEN_ARGS - -am -pl <module_name>) to control that but now this being static we lose the ability to make the module_name vary based on application.

In short does oc new-app provides a way to define build/pass environment variables as well? I think JSON/YAML files we can do so if we choose "oc create".

**Comment by gt-tech** [comment]
It seems that underlying s2i is capable of taking environment variables so seems more of a way for "oc" command to be able to receive it and then pass to underlying s2i

https://github.com/openshift/source-to-image/blob/master/docs/debugging-s2i.md#passing-in-environment-variables

**Comment by bparees** [comment]
today the new-app "-e" argument is used to add env variables to the deploymentconfig that is generated by new-app, but it does not add the env variables to the buildconfig.  you'd have to edit the buildconfig after it is created, to add the env variables.

note that there is also an "oc new-build" command which only generates a buildconfig (it does not generate a deploymentconfig or service).  oc new-build also has a "-e" argument and in that case, the env variable arguments are added to the resulting buildconfig.  So that could be another way for you to generate a correct buildconfig.

but i think it would make sense to enhance oc new-app so that you can specify env vars that get added to the buildconfig it produces, we'll leave this issue open for that.

**Comment by jupierce** [comment]
There is some debate about how to implement this while balancing two factors:
(1) Keeping new-app as simple as possible (i.e. not adding yet another command line argument to only set environment variables in the BC)
(2) Not having potentially sensitive information leak into the BC if the user does not intend for it to.

Deferring until @bparees can weigh in again.

**Comment by bparees** [comment]
I think i could ultimately be convinced to just have -e set everything everywhere all the time, and if you want more customized values, then it's up to you to create the objects and then edit them (using oc set env).  In which case we'll handle it in the trello card:
https://trello.com/c/LOfOePwR

Closing this, we'll hash it out in the trello card.
